src/main/java/com/microsoft/azure/spring/data/cosmosdb/core/DocumentDbTemplate.java

```diff
@@ -382,6 +382,9 @@ private RequestOptions getRequestOptions(PartitionKey key, Integer requestUnit)
         if (query.getSort().isSorted()) { // avoiding unnecessary query with DocumentCollection
             query.validateSort(domainClass, isCollectionSupportSortByString(getDocCollection(collectionName)));
         }
+        if (query.getCriteria().getType().equals(CriteriaType.STARTS_WITH)) {
```

Comment:
if there are two validate here, suggest to add new method include both sort and startwith validation.